spring-context/src/main/java/org/springframework/instrument/classloading/jboss/JBossLoadTimeWeaver.java

```diff
@@ -76,12 +77,25 @@ public JBossLoadTimeWeaver(@Nullable ClassLoader classLoader) {
 			}
 			transformer.setAccessible(true);
 
-			this.delegatingTransformer = transformer.get(classLoader);
-			if (!this.delegatingTransformer.getClass().getName().equals(DELEGATING_TRANSFORMER_CLASS_NAME)) {
+			Object suggestedTransformer = transformer.get(classLoader);
+
+            if (suggestedTransformer.getClass().getName().equals(WRAPPER_TRANSFORMER_CLASS_NAME)) {
+                Field wrappedTransformer = ReflectionUtils.findField(suggestedTransformer.getClass(), "transformer");
+                if (wrappedTransformer == null) {
+                    throw new IllegalArgumentException("Could not find 'transformer' field on JBoss ClassTransformer: " +
+                            suggestedTransformer.getClass().getName());
+                }
+                wrappedTransformer.setAccessible(true);
+                suggestedTransformer = wrappedTransformer.get(suggestedTransformer);
+            }
+
+            if (!suggestedTransformer.getClass().getName().equals(DELEGATING_TRANSFORMER_CLASS_NAME)) {
 				throw new IllegalStateException(
 						"Transformer not of the expected type DelegatingClassFileTransformer: " +
 						this.delegatingTransformer.getClass().getName());
```

Comment:
```suggestion
						suggestedTransformer.getClass().getName());
```
otherwise nullpointer exception